kubernetes/filters.go

```diff
@@ -240,11 +240,24 @@ func FilterRegistryServicesByServices(registryServices []*RegistryService, servi
 	return filtered
 }
 
-func FilterByRegistryService(hostname string, registryService *RegistryService) bool {
-	// Basic filter using Hostname
-	// TODO use the ExportTo, Namespace, ServiceRegistry and other attributes to refine the match
+func FilterByRegistryService(itemNamespace string, hostname string, registryService *RegistryService) bool {
+	// Basic filter using Hostname, also consider exported Namespaces of Service
 	// but for a first iteration if it's found in the registry it will be considered "valid" to reduce the number of false validation errors
-	return hostname == registryService.Hostname
+	if hostname == registryService.Hostname {
+		exportTo := registryService.IstioService.Attributes.ExportTo
+		if len(exportTo) > 0 {
+			for exportToNs := range exportTo {
+				// take only namespaces where it is exported to, or if it is exported to all namespaces
+				if exportToNs == "*" || exportToNs == itemNamespace {
```

Comment:
How this filter should work with the "." case ?

Given the example:

```
  {
    "Attributes": {
      "ServiceRegistry": "External",
      "Name": "discovery.east-mesh-system.svc.west-mesh.local",
      "Namespace": "east-mesh-system",
      "Labels": {
        "topology.istio.io/network": "network-west-mesh"
      },
      "UID": "",
      "ExportTo": {
        ".": true
      },
      "LabelSelectors": null,
      "ClusterExternalAddresses": null,
      "ClusterExternalPorts": null
    },
    "ports": [
      {
        "name": "https-discovery",
        "port": 8188,
        "protocol": "HTTPS"
      }
    ],
    "creationTime": "2021-11-10T16:17:33.262606433Z",
    "hostname": "discovery.east-mesh-system.svc.west-mesh.local",
    "address": "0.0.0.0",
    "autoAllocatedAddress": "240.240.0.1",
    "Mutex": {},
    "Resolution": 1,
    "MeshExternal": true
  },
```

I understand that this service should be visible in the condition that the `ExportTo == "."` and the namespace of the service is same of the parameter, isn't it ?

Reply:
Right, it is correct to include '.' as well, however those services are considered as services from current namespace, not imported so would not be missed and would be matched before reaching this filters.